apm-agent-core/src/test/java/co/elastic/apm/agent/impl/transaction/SameKindCompressionStrategyTest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.impl.transaction;
+
+class SameKindCompressionStrategyTest extends AbstractCompressionStrategyTest {
+
+    private int spansCreated = 0;
```

Comment:
should this be static?

Reply:
My intention was that each test start with has the same sequence of span names. But since the span name doesn't really matter (only that they are different), I can make it static if you like.

Reply:
leave it, the scenario I'm thinking of - maintenance error with reused instance - should be caught by review